neuralpredictors/layers/cores/conv2d.py

```diff
@@ -542,3 +541,165 @@ def initialize(self):
+        self._input_weights_regularizer = regularizers.__dict__[input_regularizer](**regularizer_config)
+
+        self.layers = layers
+        self.gamma_input = gamma_input
+        self.input_channels = input_channels
+        self.hidden_channels = hidden_channels
+        self.skip = skip
+        self.features = nn.Sequential()
+        self.n_se_blocks = n_se_blocks
+        if stack is None:
+            self.stack = range(self.layers)
+        else:
+            self.stack = [*range(self.layers)[stack:]] if isinstance(stack, int) else stack
+
+        # --- first layer
+        layer = OrderedDict()
+        layer["conv"] = nn.Conv2d(
+            input_channels, hidden_channels, input_kern, padding=input_kern // 2 if pad_input else 0, bias=bias
+        )
+        if batch_norm:
+            layer["norm"] = nn.BatchNorm2d(hidden_channels, momentum=momentum)
+        if (layers > 1 or final_nonlinearity) and not linear:
+            layer["nonlin"] = nn.ELU(inplace=True)
+        self.features.add_module("layer0", nn.Sequential(layer))
+
+        if not isinstance(hidden_kern, Iterable):
+            hidden_kern = [hidden_kern] * (self.layers - 1)
+
+        # --- other layers
+        for l in range(1, self.layers):
+            layer = OrderedDict()
+            hidden_padding = ((hidden_kern[l - 1] - 1) * hidden_dilation + 1) // 2
+            if depth_separable:
+                layer["ds_conv"] = DepthSeparableConv2d(
+                    hidden_channels,
+                    hidden_channels,
+                    kernel_size=hidden_kern[l - 1],
+                    dilation=hidden_dilation,
+                    padding=hidden_padding,
+                    bias=False,
+                    stride=1,
+                )
+            elif attention_conv:
+                layer["conv"] = AttentionConv(
+                    hidden_channels if not skip > 1 else min(skip, l) * hidden_channels,
+                    hidden_channels,
+                    hidden_kern[l - 1],
+                    padding=hidden_padding,
+                    bias=bias and not batch_norm,
+                )
+            else:
+                layer["conv"] = nn.Conv2d(
+                    hidden_channels if not skip > 1 else min(skip, l) * hidden_channels,
+                    hidden_channels,
+                    hidden_kern[l - 1],
+                    padding=hidden_padding,
+                    bias=bias,
+                    dilation=hidden_dilation,
+                )
+            if batch_norm:
+                layer["norm"] = nn.BatchNorm2d(hidden_channels, momentum=momentum)
+
+            if (final_nonlinearity or l < self.layers - 1) and not linear:
+                layer["nonlin"] = nn.ELU(inplace=True)
+
+            if (self.layers - l) <= self.n_se_blocks:
+                layer["seg_ex_block"] = SqueezeExcitationBlock(in_ch=hidden_channels, reduction=se_reduction)
+
+            self.features.add_module("layer{}".format(l), nn.Sequential(layer))
```

Comment:
Consider splitting these up into different methods, e.g. `add_first_layer` and `add_other_layers`